infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/NodeSelectorProvisioner.java

```diff
@@ -46,4 +47,10 @@ public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
           .forEach(d -> d.getSpec().setNodeSelector(nodeSelectorAttributes));
     }
   }
+
+  public void provision(PodSpec podSpec) {
+    if (!nodeSelectorAttributes.isEmpty()) {
```

Comment:
The above `provision` method sets the node selector attributes unconditionally, even if empty. Do we need a different behavior here?

Reply:
I may be misunderstanding what you mean here, but the above function has its streaming expression wrapped in the exact same condition that is used in this method (although GitHub does seem to be hiding those lines since they weren't modified).

Reply:
I am blind, sorry 🤦🏼